src/DummyQueryContext.cs

```diff
@@ -1,24 +1,80 @@
-﻿using System.Linq.Expressions;
+﻿using System;
+using System.Linq.Expressions;
 using Starcounter.Linq.Visitors;
 
 namespace Starcounter.Linq
 {
     public interface IQueryContext
     {
         object Execute(Expression expression);
+        object Execute<TResult>(Expression expression);
         string GetQuery(Expression expression);
     }
 
     public class QueryContext<T> : IQueryContext
     {
-        public object Execute(Expression expression)
+        public object Execute(Expression expression) => Execute<T>(expression);
+
+        public object Execute<TResult>(Expression expression)
         {
             var query = new QueryBuilder<T>();
 
             RootVisitor<T>.Instance.Visit(expression, query);
             var sql = query.BuildSqlString();
             var variables = query.GetVariables();
-            return Db.SlowSQL<T>(sql, variables); ;
+
+            if (typeof(TResult).IsGenericType)
+            {
+                return Db.SQL<T>(sql, variables);
+            }
+            else if (typeof(TResult).IsClass)
+            {
+                return Db.SlowSQL<TResult>(sql, variables);
+            }
+
+            var result = Db.SlowSQL(sql, variables).First;
+
+            if (result.GetType() != typeof(TResult))
```

Comment:
IMO we should have a comment pointing back to https://github.com/Starcounter/Home/issues/209
To explain why we do this.

Reply:
Added comment